### PR TITLE
added some different solutions + a prompt clarification for one question

### PR DESCRIPTION
**Pull request by teallarson**

### CODE REVIEW

**Review by kjesek — APPROVED** [review]
Looks great!

File: consecutive-numbers/tealsGetConsecutiveNumbers.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  if (!arr[0]) return '';
+  if (arr.length === 1) return arr[0].toString();
+
+  let result = '';
+  let i = 0;
+
+  while (i<arr.length){
+    let start = arr[i];
+    let j = i + 1;
+    let end = arr[j];
```
Comment (kjesek) [inline consecutive-numbers/tealsGetConsecutiveNumbers.js:14]: try naming your variables what they actually are. maybe firstNumber or something similar

File: consecutive-numbers/tealsGetConsecutiveNumbers.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      j++;
+      end = arr[j];
+    } 
+
+    if (arr[i] == arr[j-1]){
+      result += start.toString() + ', ';
+      i = j;
+    } else {
+      result += start.toString() + '-' + arr[j-1].toString() + ', '
+      i = j;
```
Comment (kjesek) [inline consecutive-numbers/tealsGetConsecutiveNumbers.js:26]: see if you can use the `.toString()` method a little higher up here. maybe at the initialization steps?

File: array-identity-index/tealsArrayIdentity.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+//runtime: max = O((N^2)/2) if indices array is filled with 0's min
+//space: O(1) assuming output array is a given
+//One interesting case is that sometimes there are empty spaces in the array, even if the inputs are the same length. If the values in the indices array are too high, the lower indices do not get filled.  This program returns them as just empty spaces in the array and then extends the array to move items to the right.  IE: [a, b, c, d, e], [4, 4, 4, 4, 4] ==> [null, null, null, null, e, d, c, b, a]... probably not the expected/desired behavior, but not sure how the "interviewer" would want such a case handled.
+
+const arrangeChars = (chars, indices) => {
+  if (chars.length !== indices.length) throw 'input array lengths do not match!';
+  if (chars.length === 0) return [];
+
```
Comment (kjesek) [inline array-identity-index/tealsArrayIdentity.js:8]: nice job looking for all cases

File: factorial/tealFactorial.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+//Solved recursively.  O(N) space and time if accounting for the call stack space and size
+
+const factorial = (num) => {
+  if (typeof(num) !== 'number') throw 'please input a number'
+  if (num == null) throw 'no number given!'
+  if (num === 0) return 1;
```
Comment (kjesek) [inline factorial/tealFactorial.js:6]: nice job with the errors here

File: consecutive-numbers/tealsGetConsecutiveNumbers.js (hunk 2 above)
Comment (teallarson) [reply consecutive-numbers/tealsGetConsecutiveNumbers.js:26]: Thanks.  It actually didn't need it at all... dynamic typing drives me nuts sometimes, but is helpful here!

**Pull request merged**